sourcecode/chesssimul/board.py

```diff
@@ -189,36 +195,55 @@ def move_knight(self, move: Move) -> bool:
             return False
 
     def move_bishop(self, move: Move) -> bool:
-
-        # variable test
+        
+        ##############################
+        #variable test 
         self.board[6][1].piece = None
+        ##############################
 
-        print(range(move.start[0], move.end[0]))
-        # postion de la pièce avant son déplacement dans l'échiquier
-        start_square: Square = self.board[move.start[0]][move.start[1]]
-        # postion de la pièce après son déplacement dans l'échiquier
-        end_square: Square = self.board[move.end[0]][move.end[1]]
-        # Pion sur le chemin du fou
-        for line in range(move.end[0]):
-            print("line")
-            print(line)
-            # if (move.start[0] == line):
+        if (move.end[0] >= 0 and move.end[1] <= 7):
+
+            iteration_lin: float = abs(move.end[0] - move.start[0])
+            iteration_col: float = abs(move.end[1] - move.start[1])
+
+            lin_sign: int = 1
+            col_sign: int = 1
 
-            for col in move.start[1:move.end[1]]:
-                print(col)
-                if not (self.board[line][col].isEmpty):
-                    ###Pièce sur le chemin du fou
-                    print("Une pièce bloque la diagonale du fou")
-                    break
-                elif (line == move.end[0] and col == move.end[1]):
-                    if (end_square.piece.color == start_square.piece.color):
-                        ###Pièce de même color
-                        print("prise impossible, pièce de même couleur")
-                        break
-                    if (end_square.piece.color != start_square.piece.color):
-                        self.board[move.end[0]][move.end[1]] = start_square
-        print('end')
+            if (iteration_lin == iteration_col):
+                if ((move.end[0] - move.start[0]) < 0 and (move.end[1] - move.start[1]) < 0):
```

Comment:
You can simplify further, as you don't need to check all 4 cases : 
if move.end[0] - move.start[0]  < 0: 
    lin_sign = - 1 
if move.end[1] - move.end[1] < 0 : 
    col_sign = -1